### PR TITLE
Adding functions jwt, filter, JwtUtils, AuthRequestDTO, AuthResponseDTO

### PR DESCRIPTION
**Pull request by Lucasfranco1**
**SUMMARY**

- Adding JWT
- Created filter jwt
-Authenticate success
- environment variable in SECRET_KEY and Authorization

### CODE REVIEW

**Comment by anavduo** [comment]
Looks good to me

File: src/main/java/com/alkemy/ong/auth/service/JwtUtils.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        return extractExpiration(token).before(new Date());
+    }
+
+    public String generateToken(Authentication authentication) {
+        UserDetails userDetails=(UserDetails) authentication.getPrincipal();
+        List<String> roles=userDetails.getAuthorities().stream().map(GrantedAuthority::getAuthority).collect(Collectors.toList());
+        return Jwts.builder()
+                .setSubject(userDetails.getUsername())
+                .claim("roles", roles)
+                .setIssuedAt(new Date())
```
[the remark this answers is not in the document]
Comment (Lucasfranco1) [reply src/main/java/com/alkemy/ong/auth/service/JwtUtils.java:58]: I add it with no problem.

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -13,5 +13,5 @@ app.sendgrid.key:
 app.sendgrid.from-address:
 #JWT
 jwt.secret = 
-jwt.authorization =
```
Comment (DanteIturri) [inline src/main/resources/application.properties]: added variable the enviroment, use this format ${ENVIROMENT NAME}

Comment (Lucasfranco1) [reply src/main/resources/application.properties]: Thanks, added variable the enviroment.

**Pull request merged**